### PR TITLE
Expression double power law

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Corrected the Component EELS_Double_Power_Law according to #2186 to make left_cutoff work properly
- Added Docstring.
- Added test function.
- Added gradients.

### Progress of the PR
- [X] Change implemented (can be split into several points),
- [X] update docstring (if appropriate),
- [X] add tests,
- [X] ready for review.

### Minimal example of the bug fix or the new feature
```python
>>> import hyperspy.api as hs
>>> import numpy as np
>>> s = hs.signals.Signal1D(np.arange(10))
>>> # Your new feature...
```
Note that this example can be useful to update the user guide.

### CODE REVIEW

**Comment by thomasaarholt** [comment]
Not sure why I'm getting the following when calling the following. I copied the component code directly into a notebook cell:

```python
dp = DoublePowerLaw()

Traceback (most recent call last):

  File "/Users/thomas/miniconda3/envs/py37/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3296, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-10-44535b9b65ac>", line 1, in <module>
    dp = DoublePowerLaw()

  File "<ipython-input-8-c9a37ea971a5>", line 81, in __init__
    **kwargs,

  File "/Users/thomas/hyperspy/hyperspy/_components/expression.py", line 151, in __init__
    self.compile_function(module=module, position=position)

  File "/Users/thomas/hyperspy/hyperspy/_components/expression.py", line 244, in compile_function
    dummify=False)

  File "/Users/thomas/miniconda3/envs/py37/lib/python3.7/site-packages/sympy/utilities/lambdify.py", line 773, in lambdify
    c = compile(funcstr, filename, 'exec')

  File "<lambdifygenerated-2>", line 2
    return (evaluate('  # Not supported in Python:
                                                  ^
SyntaxError: EOL while scanning string literal
```

**Comment by jlaehne** [comment]
Works for me, also if I paste the component into my notebook, like you did.

Do you have the most recent development version of hyperspy including #2186? This merge fixed the issue of having a 'where' function in the expression definition, so without it the component will not work.

**Comment by thomasaarholt** [comment]
That'll be it. I'm not sure why it didn't work, since I rebased off next_minor. But that explains the error. cheers!

File: hyperspy/_components/eels_double_power_law.py [hunk 1]
```diff
@@ -57,13 +94,41 @@ def __init__(self, A=1e-5, r=3., origin=0., shift=20., ratio=1.,
...
+                        self.shift.value) ** (-self.r.value) / (x - 
+                        self.origin.value - self.shift.value) + 
+                        self.A.value * self.r.value * (x - self.origin.value) 
+                        ** (-self.r.value) / (x - self.origin.value), 0)
+
+    def grad_shift(self, x):
+        return np.where(x > self.left_cutoff.value, self.A.value * self.r.value
+                        * self.ratio.value(x - self.origin.value - 
+                        self.shift.value) ** (-self.r.value) / 
+                        (x - self.origin.value - self.shift.value), 0)
```
Comment (ericpre) [inline hyperspy/_components/eels_double_power_law.py]: Fix typo and simplify to:
```python
        return np.where(x > self.left_cutoff.value, self.A.value * self.r.value
                        * self.ratio.value * (x - self.origin.value - 
                        self.shift.value) ** (-self.r.value - 1), 0)
```

File: hyperspy/_components/eels_double_power_law.py [hunk 2]
```diff
@@ -57,13 +94,41 @@ def __init__(self, A=1e-5, r=3., origin=0., shift=20., ratio=1.,
...
+                        self.A.value * (x - self.origin.value) ** 
+                        (-self.r.value) * np.log(x - self.origin.value), 0)
+
+    def grad_origin(self, x):
+        return np.where(x > self.left_cutoff.value, self.A.value * self.r.value
+                        * self.ratio.value * (x - self.origin.value -
+                        self.shift.value) ** (-self.r.value) / (x - 
+                        self.origin.value - self.shift.value) + 
+                        self.A.value * self.r.value * (x - self.origin.value) 
+                        ** (-self.r.value) / (x - self.origin.value), 0)
```
Comment (ericpre) [inline hyperspy/_components/eels_double_power_law.py]: Simplify to:
```python
        return np.where(x > self.left_cutoff.value, self.A.value * self.r.value
                        * self.ratio.value * (x - self.origin.value - self.shift.value) 
                        ** (-self.r.value - 1) + self.A.value * self.r.value
                        * (x - self.origin.value) ** (-self.r.value - 1), 0)
```

File: hyperspy/tests/component/test_double_power_law.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    g.r.value = 2
+    g.origin.value = 1
+    g.shift.value = 2
+    g.ratio.value = 2
+    assert np.isinf(g.function(1))
+    assert np.isinf(g.function(3))
+    assert g.function(-1) == 0
+    assert g.function(0) == 0
+    assert g.function(2) == 9
+    assert_allclose(g.function(10), 0.15948602)
```
Comment (ericpre) [inline hyperspy/tests/component/test_double_power_law.py:37]: Maybe add a call to the gradient function? It would have help to spot the typo!

**Review by ericpre — CHANGES_REQUESTED** [review]
A few comments, otherwise it looks good to merge!

File: hyperspy/_components/eels_double_power_law.py (hunk 1 above)
Comment (jlaehne) [reply hyperspy/_components/eels_double_power_law.py]: done

File: hyperspy/tests/component/test_double_power_law.py (hunk 3 above)
Comment (jlaehne) [reply hyperspy/tests/component/test_double_power_law.py:37]: done

**Review by ericpre — APPROVED** [review]

**Pull request merged**